alchemy/src/neon/database.ts

```diff
@@ -0,0 +1,152 @@
+import type { Context } from "../context.ts";
+import { Resource } from "../resource.ts";
+import { handleApiError } from "./api-error.ts";
+import { createNeonApi, type NeonApiOptions } from "./api.ts";
+
+export interface NeonDatabaseProps extends NeonApiOptions {
+  project_id: string;
+  branch_id: string;
+  name: string;
+  owner_name: string;
+  adopt?: boolean;
+}
+
+export interface NeonDatabaseType {
+  id: number;
+  branch_id: string;
+  name: string;
+  owner_name: string;
+  created_at: string;
+  updated_at: string;
+}
+
+interface NeonDatabaseApiResponse {
+  database: NeonDatabaseType;
+}
+
+export const NeonDatabase = Resource(
```

Comment:
You're not following the standards laid out in the cursorrules file. 

There must be a NeonDatabase interface that extends Resource. 

All properties and interfaces must be documented. 

Make sure to document the examples and different usage.